tests/test_metadata.py

```diff
@@ -4,16 +4,36 @@
 from nose.tools import *  # PEP8 asserts
 
 from framework.forms.utils import process_payload
+from framework.mongo import database as database_proxy, set_up_storage
+from modularodm import storage
+from modularodm.exceptions import KeyExistsException
 
+import website.models
+from website import settings
 from website.project.model import MetaSchema
 from website.project.model import ensure_schemas
 from website.project.metadata.schemas import OSF_META_SCHEMAS
 
-from tests.base import OsfTestCase
+from tests.base import OsfTestCase, teardown_database
 
 
 class TestMetaData(OsfTestCase):
 
+    def tearDown(self):
+        super(TestMetaData, self).tearDown()
+
+        # For whatever reason, DbTestCase only wipes the database during
+        # *class* setup and teardown. This leaks database state across test
+        # cases, which a) smells pretty bad, and b) breaks the test_metaschema_
+        # tests here.
+
+        teardown_database(database=database_proxy._get_current_object())
+        set_up_storage(
```

Comment:
I yanked this code straight [from](https://github.com/CenterForOpenScience/osf.io/blob/f559d4749701b213d8a12a76057b6788498b89e5/tests/base.py#L137-L144) `DbTestCase.setUpClass`. I can try splitting the two calls across `setUp` and `tearDown` here ...